### PR TITLE
Add mailer previews for Devise mailer

### LINKED ISSUE #10575: Mailer previews for DeviseInvitable mailer are absent
**Describe the bug**

Mailer previews for DeviseInvitable mailer are absent. There's demand to modify the content of those emails (#9849), so this would be good to add.

I'm happy to pick this up.

### PR DESCRIPTION
**Pull request by boardfish**
~~Submitting in draft until I've tested this locally.~~ Decided against that as it's a smallish change. Still haven't yet brought up my environment - I'll be in touch to figure that part out.

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Adds mailer previews for all Devise and DeviseInvitable mail.

## Related Tickets & Documents

Closes #10575 

## QA Instructions, Screenshots, Recordings

View the mailer previews at http://localhost:3000/rails/mailers/devise_mailer.

## Added tests?

- [ ] yes
- [x] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.forem.com
- [ ] readme
- [x] no documentation needed - already present under Previewing Emails.

### CODE REVIEW

**Review by benhalpern — APPROVED** [review]

**Comment by benhalpern** [comment]
This is great

**Review by rhymes — CHANGES_REQUESTED** [review]
Hi @boardfish ! This is not working for me, it says:

```ruby
Zeitwerk::NameError - expected file /forem/spec/mailers/previews/devise_mailer_preview.rb to define constant DeviseMailerPreview, but didn't:
   app/middlewares/set_cookie_domain.rb:13:in `call'
```

I think the issue here is in the nameing of the file. If the class is `Devise::MailerPreview` it should be located in `app/mailers/previews/devise/mailer_preview.rb` not `spec/mailers/previews/devise_mailer_preview.rb`

**Comment by boardfish** [comment]
Thanks both for checking this out for me! I still haven't been able to get this running locally, but that makes sense @rhymes - I'd hope it ought to work once it's moved.

File: spec/mailers/previews/devise/mailer_preview.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# Preview all emails at http://localhost:3000/rails/mailers/devise_mailer
+module Devise
+  # :nodoc:
+  class MailerPreview < ActionMailer::Preview
+    def confirmation_instructions
+      Devise::Mailer.confirmation_instructions(User.last, "faketoken")
+    end
+
+    def email_changed
+      Devise::Mailer.email_changed(User.last, User.last.email)
```
Comment (rhymes) [inline spec/mailers/previews/devise/mailer_preview.rb:12]: this is not working: http://localhost:3000/rails/mailers/devise/mailer/email_changed

File: spec/mailers/previews/devise/mailer_preview.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    end
+
+    def invitation_instructions
+      Devise::Mailer.email_changed(User.last, "faketoken")
```
Comment (rhymes) [inline spec/mailers/previews/devise/mailer_preview.rb:16]: this is not working http://localhost:3000/rails/mailers/devise/mailer/invitation_instructions

File: spec/mailers/previews/devise/mailer_preview.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    end
+
+    def password_changed
+      Devise::Mailer.email_changed(User.last)
+    end
+
+    def reset_password_instructions
+      Devise::Mailer.email_changed(User.last, "faketoken")
```
Comment (rhymes) [inline spec/mailers/previews/devise/mailer_preview.rb:24]: this is not working http://localhost:3000/rails/mailers/devise/mailer/reset_password_instructions

**Review by rhymes — COMMENTED** [review]
Hi @boardfish, some of these previews aren't working. As I remember you're having difficulties setting up the app, would you mind if I added the changes to make them work as I can test them locally?

**Comment by boardfish** [comment]
That'd be a quicker way to push this through for sure. Much appreciated!

**Comment by boardfish** [comment]
Thanks for fixing those up @rhymes! Looking back, I made some big oversights, so I apologize for that.

Could this be labelled `hacktoberfest-accepted` if not merged, please?

**Review by rhymes — APPROVED** [review]

**Pull request merged**